packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart/ForceFunctionStartExecutor.ts

```
@@ -0,0 +1,133 @@
+import { channelService, OUTPUT_CHANNEL } from '../../../channels';
+import { nls } from '../../../messages';
+import { notificationService } from '../../../notifications';
+import { telemetryService } from '../../../telemetry';
+import { OrgAuthInfo } from '../../../util';
+import { FunctionService } from '../functionService';
+import {
+  FUNCTION_DEFAULT_DEBUG_PORT,
+  FUNCTION_DEFAULT_PORT
+} from '../types/constants';
+
+export abstract class ForceFunctionStartExecutor extends LibraryCommandletExecutor<
+  string
+> {
+  protected UNEXPECTED_ERROR_KEY = 'force_function_start_unexpected_error';
+
+  constructor(startMessageKey: string, logName: string) {
+    super(nls.localize(startMessageKey), logName, OUTPUT_CHANNEL);
+    this.cancellable = true;
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: vscode.Progress<{
+      message?: string | undefined;
+      increment?: number | undefined;
+    }>,
+    token?: vscode.CancellationToken
+  ): Promise<boolean> {
+    const sourceFsPath = response.data;
+    const functionDirPath = FunctionService.getFunctionDir(sourceFsPath);
+    if (!functionDirPath) {
+      const warningMessage = nls.localize(
+        'force_function_start_warning_no_toml'
+      );
+      notificationService.showWarningMessage(warningMessage);
+      telemetryService.sendException(
+        'force_function_start_no_toml',
+        warningMessage
+      );
+      return false;
+    }
+
+    channelService.showChannelOutput();
+
+    try {
+      await OrgAuthInfo.getDefaultUsernameOrAlias(false).then(
+        defaultUsernameorAlias => {
+          if (!defaultUsernameorAlias) {
+            const message = nls.localize('force_function_start_no_org_auth');
+            channelService.appendLine(message);
+            channelService.showChannelOutput();
+            notificationService.showInformationMessage(message);
+          }
+        }
+      );
+    } catch (error) {
```

I don't really understand this line.  empty catches only lead to sadness 🤷